# 2- Preparar Datos:

## A) Cargar las librerias Necesarias:

In [3]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


librerias para limpieza de datos:

In [4]:
library(here)

library(skimr)

library(janitor)

library(dplyr)

here() starts at /kaggle/working


Attaching package: ‘janitor’


The following objects are masked from ‘package:stats’:

    chisq.test, fisher.test




## B) Importar Archivos Csv:

In [5]:
febrero_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202302-divvy-tripdata.csv")
marzo_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202303-divvy-tripdata.csv")
abril_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202304-divvy-tripdata.csv")
mayo_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202305-divvy-tripdata.csv")
junio_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202306-divvy-tripdata.csv")
julio_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202307-divvy-tripdata.csv")
agosto_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202308-divvy-tripdata.csv")
septiembre_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202309-divvy-tripdata.csv")
octubre_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202310-divvy-tripdata.csv")
noviembre_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202311-divvy-tripdata.csv")
diciembre_2023 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202312-divvy-tripdata.csv")
enero_2024 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202401-divvy-tripdata.csv")
febrero_2024 <- read_csv("/kaggle/input/cyclist-datos-de-uso/202402-divvy-tripdata.csv")

Rows: 190445 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (7): ride_id, rideable_type, start_station_name, start_station_id, end_...
dbl  (4): start_lat, start_lng, end_lat, end_lng
dttm (2): started_at, ended_at

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 258678 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (7): ride_id, rideable_type, start_station_name, start_station_id, end_...
dbl  (4): start_lat, start_lng, end_lat, end_lng
dttm (2): started_at, ended_at

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 426590 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (7

Previsualizar los datos:

para ver los primeros 6 registros, usaré la funcion head() y seleccionare uno de los datasets


In [6]:
head(febrero_2023)

ride_id          rideable_type started_at          ended_at           
1 CBCD0D7777F0E45F classic_bike  2023-02-14 11:59:42 2023-02-14 12:13:38
2 F3EC5FCE5FF39DE9 electric_bike 2023-02-15 13:53:48 2023-02-15 13:59:08
3 E54C1F27FA9354FF classic_bike  2023-02-19 11:10:57 2023-02-19 11:35:01
4 3D561E04F739CC45 electric_bike 2023-02-26 16:12:05 2023-02-26 16:39:55
5 0CB4B4D53B2DBE05 electric_bike 2023-02-20 11:55:23 2023-02-20 12:05:48
6 C67EB62172C472EB classic_bike  2023-02-24 18:50:16 2023-02-24 18:56:40
  start_station_name           start_station_id end_station_name             
1 Southport Ave & Clybourn Ave TA1309000030     Clark St & Schiller St       
2 Clarendon Ave & Gordon Ter   13379            Sheridan Rd & Lawrence Ave   
3 Southport Ave & Clybourn Ave TA1309000030     Aberdeen St & Monroe St      
4 Southport Ave & Clybourn Ave TA1309000030     Franklin St & Adams St (Temp)
5 Prairie Ave & Garfield Blvd  TA1307000160     Cottage Grove Ave & 63rd St  
6 Wells St & Concord Ln        TA1308000050     Clybourn Ave & Division St   
  end_station_id start_lat start_lng end_lat  end_lng   member_casual
1 TA1309000024   41.92077  -87.66371 41.90799 -87.63150 casual       
2 TA1309000041   41.95788  -87.64958 41.96952 -87.65469 casual       
3 13156          41.92077  -87.66371 41.88042 -87.65552 member       
4 TA1309000008   41.92087  -87.66373 41.87943 -87.63550 member       
5 KA1503000054   41.79483  -87.61879 41.78053 -87.60597 member       
6 TA1307000115   41.91213  -87.63466 41.90461 -87.64055 member

Información sobre la estructura dle dataframe:

In [7]:
str(febrero_2023)

spc_tbl_ [190,445 × 13] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ ride_id           : chr [1:190445] "CBCD0D7777F0E45F" "F3EC5FCE5FF39DE9" "E54C1F27FA9354FF" "3D561E04F739CC45" ...
 $ rideable_type     : chr [1:190445] "classic_bike" "electric_bike" "classic_bike" "electric_bike" ...
 $ started_at        : POSIXct[1:190445], format: "2023-02-14 11:59:42" "2023-02-15 13:53:48" ...
 $ ended_at          : POSIXct[1:190445], format: "2023-02-14 12:13:38" "2023-02-15 13:59:08" ...
 $ start_station_name: chr [1:190445] "Southport Ave & Clybourn Ave" "Clarendon Ave & Gordon Ter" "Southport Ave & Clybourn Ave" "Southport Ave & Clybourn Ave" ...
 $ start_station_id  : chr [1:190445] "TA1309000030" "13379" "TA1309000030" "TA1309000030" ...
 $ end_station_name  : chr [1:190445] "Clark St & Schiller St" "Sheridan Rd & Lawrence Ave" "Aberdeen St & Monroe St" "Franklin St & Adams St (Temp)" ...
 $ end_station_id    : chr [1:190445] "TA1309000024" "TA1309000041" "13156" "TA1309000008" ...
 $ start_l

Obvservamos un total de 190.445 registros y 13 columnas

# 3- Procesar Datos: 

## A) Unificar tablas en un DataFrame:

In [8]:
tabla_general_1 <- rbind(febrero_2023, marzo_2023, abril_2023, mayo_2023, junio_2023,
                         julio_2023, agosto_2023, septiembre_2023, octubre_2023,
                         noviembre_2023, diciembre_2023, enero_2024, febrero_2024)

## B) Previsualizar los datos:
Se omite utilizar la funcion View(), por el espacio en la notebook, pero se destaca como un metodo para visualizar  completamente el dataframe

* Previsualizacion de los primeros  6 registros:

In [9]:
head(tabla_general_1)

ride_id          rideable_type started_at          ended_at           
1 CBCD0D7777F0E45F classic_bike  2023-02-14 11:59:42 2023-02-14 12:13:38
2 F3EC5FCE5FF39DE9 electric_bike 2023-02-15 13:53:48 2023-02-15 13:59:08
3 E54C1F27FA9354FF classic_bike  2023-02-19 11:10:57 2023-02-19 11:35:01
4 3D561E04F739CC45 electric_bike 2023-02-26 16:12:05 2023-02-26 16:39:55
5 0CB4B4D53B2DBE05 electric_bike 2023-02-20 11:55:23 2023-02-20 12:05:48
6 C67EB62172C472EB classic_bike  2023-02-24 18:50:16 2023-02-24 18:56:40
  start_station_name           start_station_id end_station_name             
1 Southport Ave & Clybourn Ave TA1309000030     Clark St & Schiller St       
2 Clarendon Ave & Gordon Ter   13379            Sheridan Rd & Lawrence Ave   
3 Southport Ave & Clybourn Ave TA1309000030     Aberdeen St & Monroe St      
4 Southport Ave & Clybourn Ave TA1309000030     Franklin St & Adams St (Temp)
5 Prairie Ave & Garfield Blvd  TA1307000160     Cottage Grove Ave & 63rd St  
6 Wells St & Concord Ln        TA1308000050     Clybourn Ave & Division St   
  end_station_id start_lat start_lng end_lat  end_lng   member_casual
1 TA1309000024   41.92077  -87.66371 41.90799 -87.63150 casual       
2 TA1309000041   41.95788  -87.64958 41.96952 -87.65469 casual       
3 13156          41.92077  -87.66371 41.88042 -87.65552 member       
4 TA1309000008   41.92087  -87.66373 41.87943 -87.63550 member       
5 KA1503000054   41.79483  -87.61879 41.78053 -87.60597 member       
6 TA1307000115   41.91213  -87.63466 41.90461 -87.64055 member

* Previsualizacion de los nombres de las columnas.

In [10]:
colnames(tabla_general_1)

[1] "ride_id"            "rideable_type"      "started_at"        
 [4] "ended_at"           "start_station_name" "start_station_id"  
 [7] "end_station_name"   "end_station_id"     "start_lat"         
[10] "start_lng"          "end_lat"            "end_lng"           
[13] "member_casual"

verificamos que hay un total de 13 columnas diferentes

* Informacion sobre la estructura del data frame:

In [11]:
str(tabla_general_1)

spc_tbl_ [5,897,613 × 13] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ ride_id           : chr [1:5897613] "CBCD0D7777F0E45F" "F3EC5FCE5FF39DE9" "E54C1F27FA9354FF" "3D561E04F739CC45" ...
 $ rideable_type     : chr [1:5897613] "classic_bike" "electric_bike" "classic_bike" "electric_bike" ...
 $ started_at        : POSIXct[1:5897613], format: "2023-02-14 11:59:42" "2023-02-15 13:53:48" ...
 $ ended_at          : POSIXct[1:5897613], format: "2023-02-14 12:13:38" "2023-02-15 13:59:08" ...
 $ start_station_name: chr [1:5897613] "Southport Ave & Clybourn Ave" "Clarendon Ave & Gordon Ter" "Southport Ave & Clybourn Ave" "Southport Ave & Clybourn Ave" ...
 $ start_station_id  : chr [1:5897613] "TA1309000030" "13379" "TA1309000030" "TA1309000030" ...
 $ end_station_name  : chr [1:5897613] "Clark St & Schiller St" "Sheridan Rd & Lawrence Ave" "Aberdeen St & Monroe St" "Franklin St & Adams St (Temp)" ...
 $ end_station_id    : chr [1:5897613] "TA1309000024" "TA1309000041" "13156" "TA1309000008" ...


Se verifica que el data frame contiene 5.897.613 registros  y 13 columnas

## C) Eliminar duplicados:

Seleccionar solo aquellos registros que NO esten duplicados:

In [12]:
datos_no_duplicados <- distinct(tabla_general_1)

Verificar la cantidad de registros resultantes:

In [13]:
str(datos_no_duplicados)

tibble [5,897,613 × 13] (S3: tbl_df/tbl/data.frame)
 $ ride_id           : chr [1:5897613] "CBCD0D7777F0E45F" "F3EC5FCE5FF39DE9" "E54C1F27FA9354FF" "3D561E04F739CC45" ...
 $ rideable_type     : chr [1:5897613] "classic_bike" "electric_bike" "classic_bike" "electric_bike" ...
 $ started_at        : POSIXct[1:5897613], format: "2023-02-14 11:59:42" "2023-02-15 13:53:48" ...
 $ ended_at          : POSIXct[1:5897613], format: "2023-02-14 12:13:38" "2023-02-15 13:59:08" ...
 $ start_station_name: chr [1:5897613] "Southport Ave & Clybourn Ave" "Clarendon Ave & Gordon Ter" "Southport Ave & Clybourn Ave" "Southport Ave & Clybourn Ave" ...
 $ start_station_id  : chr [1:5897613] "TA1309000030" "13379" "TA1309000030" "TA1309000030" ...
 $ end_station_name  : chr [1:5897613] "Clark St & Schiller St" "Sheridan Rd & Lawrence Ave" "Aberdeen St & Monroe St" "Franklin St & Adams St (Temp)" ...
 $ end_station_id    : chr [1:5897613] "TA1309000024" "TA1309000041" "13156" "TA1309000008" ...
 $ start_lat  

Se verifican la misma cantidad de registros, siendo un total de 5.897.613, por lo cual no hay duplicados

## D) Seleccionar las columnas que me son utiles:
Para mi analisis, voy a utilizar 10 de las 13 columnas:

In [14]:
tabla_general_2 <- select(tabla_general_1, rideable_type, started_at, ended_at, 
                          start_station_name, end_station_name, start_lat, 
                          start_lng, end_lat, end_lng, member_casual)

Previsualizar el dataframe:

In [15]:
head(tabla_general_2)

rideable_type started_at          ended_at           
1 classic_bike  2023-02-14 11:59:42 2023-02-14 12:13:38
2 electric_bike 2023-02-15 13:53:48 2023-02-15 13:59:08
3 classic_bike  2023-02-19 11:10:57 2023-02-19 11:35:01
4 electric_bike 2023-02-26 16:12:05 2023-02-26 16:39:55
5 electric_bike 2023-02-20 11:55:23 2023-02-20 12:05:48
6 classic_bike  2023-02-24 18:50:16 2023-02-24 18:56:40
  start_station_name           end_station_name              start_lat
1 Southport Ave & Clybourn Ave Clark St & Schiller St        41.92077 
2 Clarendon Ave & Gordon Ter   Sheridan Rd & Lawrence Ave    41.95788 
3 Southport Ave & Clybourn Ave Aberdeen St & Monroe St       41.92077 
4 Southport Ave & Clybourn Ave Franklin St & Adams St (Temp) 41.92087 
5 Prairie Ave & Garfield Blvd  Cottage Grove Ave & 63rd St   41.79483 
6 Wells St & Concord Ln        Clybourn Ave & Division St    41.91213 
  start_lng end_lat  end_lng   member_casual
1 -87.66371 41.90799 -87.63150 casual       
2 -87.64958 41.96952 -87.65469 casual       
3 -87.66371 41.88042 -87.65552 member       
4 -87.66373 41.87943 -87.63550 member       
5 -87.61879 41.78053 -87.60597 member       
6 -87.63466 41.90461 -87.64055 member

## E) Renombrar columnas
Renombro las columnas por su equivalente al español:

In [16]:
tabla_general_3 <- rename(tabla_general_2,
                          tipo_bicicleta = rideable_type, 
                          fecha_hora_inicio = started_at, 
                          fecha_hora_fin = ended_at, 
                          estacion_inicio = start_station_name, 
                          estacion_fin = end_station_name, 
                          latitud_inicio = start_lat, 
                          longitud_inicio = start_lng, 
                          latitud_fin = end_lat, 
                          longitud_fin = end_lng, 
                          tipo_cliente= member_casual)

Previsualizar el dataframe:

In [17]:
head(tabla_general_3)

tipo_bicicleta fecha_hora_inicio   fecha_hora_fin     
1 classic_bike   2023-02-14 11:59:42 2023-02-14 12:13:38
2 electric_bike  2023-02-15 13:53:48 2023-02-15 13:59:08
3 classic_bike   2023-02-19 11:10:57 2023-02-19 11:35:01
4 electric_bike  2023-02-26 16:12:05 2023-02-26 16:39:55
5 electric_bike  2023-02-20 11:55:23 2023-02-20 12:05:48
6 classic_bike   2023-02-24 18:50:16 2023-02-24 18:56:40
  estacion_inicio              estacion_fin                  latitud_inicio
1 Southport Ave & Clybourn Ave Clark St & Schiller St        41.92077      
2 Clarendon Ave & Gordon Ter   Sheridan Rd & Lawrence Ave    41.95788      
3 Southport Ave & Clybourn Ave Aberdeen St & Monroe St       41.92077      
4 Southport Ave & Clybourn Ave Franklin St & Adams St (Temp) 41.92087      
5 Prairie Ave & Garfield Blvd  Cottage Grove Ave & 63rd St   41.79483      
6 Wells St & Concord Ln        Clybourn Ave & Division St    41.91213      
  longitud_inicio latitud_fin longitud_fin tipo_cliente
1 -87.66371       41.90799    -87.63150    casual      
2 -87.64958       41.96952    -87.65469    casual      
3 -87.66371       41.88042    -87.65552    member      
4 -87.66373       41.87943    -87.63550    member      
5 -87.61879       41.78053    -87.60597    member      
6 -87.63466       41.90461    -87.64055    member

## F) Ordenar por fecha

In [18]:
tabla_general_4 <- arrange(tabla_general_3,fecha_hora_inicio )

Previsualizacion:

In [19]:
head(tabla_general_4)

tipo_bicicleta fecha_hora_inicio   fecha_hora_fin     
1 electric_bike  2023-02-01 00:01:34 2023-02-01 00:25:25
2 classic_bike   2023-02-01 00:03:23 2023-02-01 00:08:42
3 classic_bike   2023-02-01 00:05:34 2023-02-01 00:12:17
4 electric_bike  2023-02-01 00:11:07 2023-02-01 00:17:40
5 electric_bike  2023-02-01 00:11:14 2023-02-01 00:15:23
6 electric_bike  2023-02-01 00:16:04 2023-02-01 00:20:37
  estacion_inicio          estacion_fin           latitud_inicio
1 NA                       NA                     41.92000      
2 University Ave & 57th St Ellis Ave & 60th St    41.79148      
3 Broadway & Ridge Ave     Broadway & Berwyn Ave  41.98404      
4 NA                       Marine Dr & Ainslie St 41.98000      
5 NA                       NA                     41.90000      
6 NA                       NA                     41.79000      
  longitud_inicio latitud_fin longitud_fin tipo_cliente
1 -87.69000       41.92000    -87.69000    member      
2 -87.59986       41.78510    -87.60107    casual      
3 -87.66027       41.97835    -87.65975    member      
4 -87.67000       41.97160    -87.65015    member      
5 -87.70000       41.91000    -87.71000    member      
6 -87.60000       41.79000    -87.59000    member

## G) verificar  las clasificaciones de los tipos de biciletas y de clientes

In [20]:
unique(tabla_general_4$tipo_bicicleta)

[1] "electric_bike" "classic_bike"  "docked_bike"

los registros indican que hay tres tipos de biciletas:"electric_bike","classic_bike" y  "docked_bike"

In [21]:
unique(tabla_general_4$tipo_cliente)

[1] "member" "casual"

los registros indican que hay dos tipos de clientes: "member" y "casual"

## H) Remplazar los valores:
Reemplace los valores al español de las columnas  tipo_bicicleta y tipo_cliente

In [22]:
tabla_general_5 <- tabla_general_4 %>%
  mutate(tipo_bicicleta = if_else(tipo_bicicleta == "classic_bike", "bicicleta clásica", tipo_bicicleta)) %>%
  mutate(tipo_bicicleta = if_else(tipo_bicicleta == "electric_bike", "bicicleta eléctrica", tipo_bicicleta)) %>%
  mutate(tipo_bicicleta = if_else(tipo_bicicleta == "docked_bike", "bicicleta docked", tipo_bicicleta))  %>%
  mutate(tipo_cliente = ifelse(tipo_cliente == "member", "miembro", tipo_cliente))


Visualizacion:

In [23]:
View(tabla_general_5)

tipo_bicicleta      fecha_hora_inicio   fecha_hora_fin     
1       bicicleta eléctrica 2023-02-01 00:01:34 2023-02-01 00:25:25
2       bicicleta clásica   2023-02-01 00:03:23 2023-02-01 00:08:42
3       bicicleta clásica   2023-02-01 00:05:34 2023-02-01 00:12:17
4       bicicleta eléctrica 2023-02-01 00:11:07 2023-02-01 00:17:40
5       bicicleta eléctrica 2023-02-01 00:11:14 2023-02-01 00:15:23
6       bicicleta eléctrica 2023-02-01 00:16:04 2023-02-01 00:20:37
7       bicicleta clásica   2023-02-01 00:17:21 2023-02-01 00:20:14
8       bicicleta clásica   2023-02-01 00:22:19 2023-02-01 00:25:07
9       bicicleta eléctrica 2023-02-01 00:25:54 2023-02-01 00:28:29
10      bicicleta eléctrica 2023-02-01 00:31:39 2023-02-01 00:44:44
11      bicicleta clásica   2023-02-01 00:33:46 2023-02-01 00:35:37
12      bicicleta eléctrica 2023-02-01 00:35:43 2023-02-01 00:37:47
13      bicicleta eléctrica 2023-02-01 00:39:41 2023-02-01 00:41:30
14      bicicleta eléctrica 2023-02-01 00:39:47 2023-02-01 00:52:49
15      bicicleta clásica   2023-02-01 00:42:25 2023-02-01 01:48:34
16      bicicleta clásica   2023-02-01 00:50:36 2023-02-01 01:05:35
17      bicicleta eléctrica 2023-02-01 01:00:16 2023-02-01 01:04:28
18      bicicleta eléctrica 2023-02-01 01:07:45 2023-02-01 01:18:45
19      bicicleta eléctrica 2023-02-01 01:09:18 2023-02-01 01:30:12
20      bicicleta eléctrica 2023-02-01 01:09:32 2023-02-01 01:18:20
21      bicicleta eléctrica 2023-02-01 01:16:50 2023-02-01 01:19:19
22      bicicleta docked    2023-02-01 01:22:35 2023-02-01 01:37:44
23      bicicleta clásica   2023-02-01 01:23:37 2023-02-01 01:27:24
24      bicicleta eléctrica 2023-02-01 01:30:31 2023-02-01 01:56:02
25      bicicleta docked    2023-02-01 01:39:55 2023-02-01 01:49:01
26      bicicleta clásica   2023-02-01 01:45:18 2023-02-01 02:06:16
27      bicicleta eléctrica 2023-02-01 01:49:50 2023-02-01 02:22:38
28      bicicleta clásica   2023-02-01 01:51:56 2023-02-01 02:59:02
29      bicicleta eléctrica 2023-02-01 01:59:28 2023-02-01 02:07:00
30      bicicleta eléctrica 2023-02-01 02:43:05 2023-02-01 02:53:19
⋮       ⋮                   ⋮                   ⋮                  
5897584 bicicleta clásica   2024-02-29 23:40:44 2024-02-29 23:59:56
5897585 bicicleta clásica   2024-02-29 23:41:46 2024-02-29 23:46:35
5897586 bicicleta clásica   2024-02-29 23:42:12 2024-02-29 23:46:44
5897587 bicicleta eléctrica 2024-02-29 23:42:35 2024-02-29 23:47:23
5897588 bicicleta clásica   2024-02-29 23:43:26 2024-02-29 23:56:49
5897589 bicicleta clásica   2024-02-29 23:43:30 2024-02-29 23:52:44
5897590 bicicleta eléctrica 2024-02-29 23:45:03 2024-02-29 23:49:54
5897591 bicicleta clásica   2024-02-29 23:45:08 2024-02-29 23:56:01
5897592 bicicleta eléctrica 2024-02-29 23:47:00 2024-02-29 23:50:17
5897593 bicicleta clásica   2024-02-29 23:47:21 2024-02-29 23:50:25
5897594 bicicleta eléctrica 2024-02-29 23:48:19 2024-02-29 23:52:39
5897595 bicicleta clásica   2024-02-29 23:48:49 2024-02-29 23:58:33
5897596 bicicleta clásica   2024-02-29 23:48:51 2024-02-29 23:55:25
5897597 bicicleta clásica   2024-02-29 23:49:31 2024-02-29 23:57:41
5897598 bicicleta clásica   2024-02-29 23:49:42 2024-02-29 23:58:00
5897599 bicicleta clásica   2024-02-29 23:50:23 2024-02-29 23:55:25
5897600 bicicleta clásica   2024-02-29 23:50:33 2024-03-01 00:06:49
5897601 bicicleta clásica   2024-02-29 23:51:47 2024-02-29 23:53:31
5897602 bicicleta eléctrica 2024-02-29 23:52:01 2024-02-29 23:54:37
5897603 bicicleta clásica   2024-02-29 23:52:45 2024-02-29 23:55:43
5897604 bicicleta clásica   2024-02-29 23:52:50 2024-03-01 00:05:20
5897605 bicicleta clásica   2024-02-29 23:54:00 2024-02-29 23:59:22
5897606 bicicleta clásica   2024-02-29 23:54:07 2024-02-29 23:59:29
5897607 bicicleta eléctrica 2024-02-29 23:55:00 2024-02-29 23:58:30
5897608 bicicleta clásica   2024-02-29 23:55:52 2024-02-29 23:59:20
5897609 bicicleta clásica   2024-02-29 23:55:54 2024-03-01 00:00:14
5897610 bicicleta eléctrica 2024-02-29 23:57:16 2024-03-01 00:01

## I) Filtrar datos:
Como el analisis busca centrarse en los miembros casuales, filtrare los registros, segun el tipo de cliente (tipo_cliente)

In [24]:
tabla_general_6 <- filter(tabla_general_5,tabla_general_5$tipo_cliente == 'casual')

previsualizar los datos del dataset resultante:

In [25]:
str(tabla_general_6)

tibble [2,090,794 × 10] (S3: tbl_df/tbl/data.frame)
 $ tipo_bicicleta   : chr [1:2090794] "bicicleta clásica" "bicicleta eléctrica" "bicicleta eléctrica" "bicicleta docked" ...
 $ fecha_hora_inicio: POSIXct[1:2090794], format: "2023-02-01 00:03:23" "2023-02-01 00:31:39" ...
 $ fecha_hora_fin   : POSIXct[1:2090794], format: "2023-02-01 00:08:42" "2023-02-01 00:44:44" ...
 $ estacion_inicio  : chr [1:2090794] "University Ave & 57th St" "Kostner Ave & Wrightwood Ave" NA "Leavitt St & Armitage Ave" ...
 $ estacion_fin     : chr [1:2090794] "Ellis Ave & 60th St" NA NA "Racine Ave & Fullerton Ave (Temp)" ...
 $ latitud_inicio   : num [1:2090794] 41.8 41.9 41.8 41.9 41.9 ...
 $ longitud_inicio  : num [1:2090794] -87.6 -87.7 -87.6 -87.7 -87.7 ...
 $ latitud_fin      : num [1:2090794] 41.8 41.9 41.8 41.9 41.9 ...
 $ longitud_fin     : num [1:2090794] -87.6 -87.8 -87.6 -87.7 -87.6 ...
 $ tipo_cliente     : chr [1:2090794] "casual" "casual" "casual" "casual" ...


vemos que el data set, se redujo a un total de 2.090.794 registros

## J) Eliminar las filas vacias o que contengan NA:

* Eliminar filas vacias:

In [26]:
tabla_general_7 <- remove_empty(tabla_general_6, "rows")

- Eliminar las que contengan NA:

In [27]:
tabla_general_8 <- drop_na(tabla_general_7)

previsualizacion de datos

In [28]:
str(tabla_general_8)

tibble [1,558,087 × 10] (S3: tbl_df/tbl/data.frame)
 $ tipo_bicicleta   : chr [1:1558087] "bicicleta clásica" "bicicleta docked" "bicicleta docked" "bicicleta clásica" ...
 $ fecha_hora_inicio: POSIXct[1:1558087], format: "2023-02-01 00:03:23" "2023-02-01 01:22:35" ...
 $ fecha_hora_fin   : POSIXct[1:1558087], format: "2023-02-01 00:08:42" "2023-02-01 01:37:44" ...
 $ estacion_inicio  : chr [1:1558087] "University Ave & 57th St" "Leavitt St & Armitage Ave" "Racine Ave & Fullerton Ave (Temp)" "Wabash Ave & Roosevelt Rd" ...
 $ estacion_fin     : chr [1:1558087] "Ellis Ave & 60th St" "Racine Ave & Fullerton Ave (Temp)" "Clark St & Wrightwood Ave" "Wabash Ave & 9th St" ...
 $ latitud_inicio   : num [1:1558087] 41.8 41.9 41.9 41.9 41.9 ...
 $ longitud_inicio  : num [1:1558087] -87.6 -87.7 -87.7 -87.6 -87.6 ...
 $ latitud_fin      : num [1:1558087] 41.8 41.9 41.9 41.9 41.9 ...
 $ longitud_fin     : num [1:1558087] -87.6 -87.7 -87.6 -87.6 -87.6 ...
 $ tipo_cliente     : chr [1:1558087] "casu

quedan un total de 1.558.087 registros para el analisis

## K) Calcular la duración de cada viaje:
Crearé una nueva columna para calcular la duracion de cada viaje:

In [42]:
tabla_general_9 <- tabla_general_8 %>%
    mutate(duracion_viaje = (fecha_hora_fin - fecha_hora_inicio))

Previsualizó los primeros 6 registros, para verificar que  la duracion se haya calculado correctamente

In [48]:
head(tabla_general_9)

tipo_bicicleta      fecha_hora_inicio   fecha_hora_fin     
1 bicicleta clásica   2023-02-01 00:03:23 2023-02-01 00:08:42
2 bicicleta docked    2023-02-01 01:22:35 2023-02-01 01:37:44
3 bicicleta docked    2023-02-01 01:39:55 2023-02-01 01:49:01
4 bicicleta clásica   2023-02-01 04:53:51 2023-02-01 04:55:57
5 bicicleta eléctrica 2023-02-01 04:56:34 2023-02-01 05:02:02
6 bicicleta clásica   2023-02-01 05:36:31 2023-02-01 05:52:48
  estacion_inicio                   estacion_fin                     
1 University Ave & 57th St          Ellis Ave & 60th St              
2 Leavitt St & Armitage Ave         Racine Ave & Fullerton Ave (Temp)
3 Racine Ave & Fullerton Ave (Temp) Clark St & Wrightwood Ave        
4 Wabash Ave & Roosevelt Rd         Wabash Ave & 9th St              
5 Wabash Ave & 9th St               Michigan Ave & Washington St     
6 Sheridan Rd & Buena Ave           Southport Ave & Roscoe St        
  latitud_inicio longitud_inicio latitud_fin longitud_fin tipo_cliente
1 41.79148       -87.59986       41.78510    -87.60107    casual      
2 41.91781       -87.68244       41.92551    -87.65840    casual      
3 41.92551       -87.65840       41.92955    -87.64312    casual      
4 41.86723       -87.62596       41.87077    -87.62573    casual      
5 41.87059       -87.62575       41.88398    -87.62468    casual      
6 41.95849       -87.65497       41.94374    -87.66402    casual      
  duracion_viaje
1 319 secs      
2 909 secs      
3 546 secs      
4 126 secs      
5 328 secs      
6 977 secs

Verifico si hay registros donde la duracion del viaje sea igual o menor a 0:

In [62]:
filtrados <- filter(tabla_general_9, duracion_viaje <= 0 )

head(filtrados)

str(filtrados)

tipo_bicicleta      fecha_hora_inicio   fecha_hora_fin     
1 bicicleta eléctrica 2023-02-17 16:42:09 2023-02-17 16:42:09
2 bicicleta eléctrica 2023-02-26 14:04:36 2023-02-26 14:04:36
3 bicicleta eléctrica 2023-03-14 16:55:24 2023-03-14 16:55:24
4 bicicleta eléctrica 2023-03-19 17:13:10 2023-03-19 17:13:10
5 bicicleta eléctrica 2023-03-26 14:53:07 2023-03-26 14:53:07
6 bicicleta eléctrica 2023-03-27 17:36:47 2023-03-27 17:36:47
  estacion_inicio                     estacion_fin                       
1 Fairbanks St & Superior St          Fairbanks St & Superior St         
2 Michigan Ave & Washington St        Michigan Ave & Washington St       
3 Mies van der Rohe Way & Chicago Ave Mies van der Rohe Way & Chicago Ave
4 Wabash Ave & Adams St               Wabash Ave & Adams St              
5 Marine Dr & Ainslie St              Marine Dr & Ainslie St             
6 Loomis St & Lexington St            Loomis St & Lexington St           
  latitud_inicio longitud_inicio latitud_fin longitud_fin tipo_cliente
1 41.89582       -87.62027       41.89575    -87.62010    casual      
2 41.88392       -87.62483       41.88398    -87.62468    casual      
3 41.89681       -87.62172       41.89694    -87.62176    casual      
4 41.87944       -87.62563       41.87947    -87.62569    casual      
5 41.97162       -87.65025       41.97160    -87.65015    casual      
6 41.87217       -87.66135       41.87223    -87.66136    casual      
  duracion_viaje
1 0 secs        
2 0 secs        
3 0 secs        
4 0 secs        
5 0 secs        
6 0 secs

tibble [227 × 11] (S3: tbl_df/tbl/data.frame)
 $ tipo_bicicleta   : chr [1:227] "bicicleta eléctrica" "bicicleta eléctrica" "bicicleta eléctrica" "bicicleta eléctrica" ...
 $ fecha_hora_inicio: POSIXct[1:227], format: "2023-02-17 16:42:09" "2023-02-26 14:04:36" ...
 $ fecha_hora_fin   : POSIXct[1:227], format: "2023-02-17 16:42:09" "2023-02-26 14:04:36" ...
 $ estacion_inicio  : chr [1:227] "Fairbanks St & Superior St" "Michigan Ave & Washington St" "Mies van der Rohe Way & Chicago Ave" "Wabash Ave & Adams St" ...
 $ estacion_fin     : chr [1:227] "Fairbanks St & Superior St" "Michigan Ave & Washington St" "Mies van der Rohe Way & Chicago Ave" "Wabash Ave & Adams St" ...
 $ latitud_inicio   : num [1:227] 41.9 41.9 41.9 41.9 42 ...
 $ longitud_inicio  : num [1:227] -87.6 -87.6 -87.6 -87.6 -87.7 ...
 $ latitud_fin      : num [1:227] 41.9 41.9 41.9 41.9 42 ...
 $ longitud_fin     : num [1:227] -87.6 -87.6 -87.6 -87.6 -87.7 ...
 $ tipo_cliente     : chr [1:227] "casual" "casual" "casual" "

Hay un total de 227 registros con una duracion de 0 segundos o que tienen duracion negativa, lo cual no es logico y deben ser descartados. Por lo que proceso a filtrar los datos de la tabla_general_9:

In [63]:
tabla_general_10 <- filter(tabla_general_9, duracion_viaje > 0 )

head(tabla_general_10)

str(tabla_general_10)

tipo_bicicleta      fecha_hora_inicio   fecha_hora_fin     
1 bicicleta clásica   2023-02-01 00:03:23 2023-02-01 00:08:42
2 bicicleta docked    2023-02-01 01:22:35 2023-02-01 01:37:44
3 bicicleta docked    2023-02-01 01:39:55 2023-02-01 01:49:01
4 bicicleta clásica   2023-02-01 04:53:51 2023-02-01 04:55:57
5 bicicleta eléctrica 2023-02-01 04:56:34 2023-02-01 05:02:02
6 bicicleta clásica   2023-02-01 05:36:31 2023-02-01 05:52:48
  estacion_inicio                   estacion_fin                     
1 University Ave & 57th St          Ellis Ave & 60th St              
2 Leavitt St & Armitage Ave         Racine Ave & Fullerton Ave (Temp)
3 Racine Ave & Fullerton Ave (Temp) Clark St & Wrightwood Ave        
4 Wabash Ave & Roosevelt Rd         Wabash Ave & 9th St              
5 Wabash Ave & 9th St               Michigan Ave & Washington St     
6 Sheridan Rd & Buena Ave           Southport Ave & Roscoe St        
  latitud_inicio longitud_inicio latitud_fin longitud_fin tipo_cliente
1 41.79148       -87.59986       41.78510    -87.60107    casual      
2 41.91781       -87.68244       41.92551    -87.65840    casual      
3 41.92551       -87.65840       41.92955    -87.64312    casual      
4 41.86723       -87.62596       41.87077    -87.62573    casual      
5 41.87059       -87.62575       41.88398    -87.62468    casual      
6 41.95849       -87.65497       41.94374    -87.66402    casual      
  duracion_viaje
1 319 secs      
2 909 secs      
3 546 secs      
4 126 secs      
5 328 secs      
6 977 secs

tibble [1,557,860 × 11] (S3: tbl_df/tbl/data.frame)
 $ tipo_bicicleta   : chr [1:1557860] "bicicleta clásica" "bicicleta docked" "bicicleta docked" "bicicleta clásica" ...
 $ fecha_hora_inicio: POSIXct[1:1557860], format: "2023-02-01 00:03:23" "2023-02-01 01:22:35" ...
 $ fecha_hora_fin   : POSIXct[1:1557860], format: "2023-02-01 00:08:42" "2023-02-01 01:37:44" ...
 $ estacion_inicio  : chr [1:1557860] "University Ave & 57th St" "Leavitt St & Armitage Ave" "Racine Ave & Fullerton Ave (Temp)" "Wabash Ave & Roosevelt Rd" ...
 $ estacion_fin     : chr [1:1557860] "Ellis Ave & 60th St" "Racine Ave & Fullerton Ave (Temp)" "Clark St & Wrightwood Ave" "Wabash Ave & 9th St" ...
 $ latitud_inicio   : num [1:1557860] 41.8 41.9 41.9 41.9 41.9 ...
 $ longitud_inicio  : num [1:1557860] -87.6 -87.7 -87.7 -87.6 -87.6 ...
 $ latitud_fin      : num [1:1557860] 41.8 41.9 41.9 41.9 41.9 ...
 $ longitud_fin     : num [1:1557860] -87.6 -87.7 -87.6 -87.6 -87.6 ...
 $ tipo_cliente     : chr [1:1557860] "casu

Verificamos que de un total de 1.558.087, quedaron 1.557.860, descontando los 227 registros que canalizamos anteriormente

## L) Estandarizar el formato de la columna duracion_viaje, actualmente esta en segundos y lo pasare a formato Hora, minutos y segundos (HH:MM:SS):

1. Convertí los datos de segundos a periodo (H:M:S):

In [64]:
tabla_general_10 <- tabla_general_10 %>%
  mutate(duracion_viaje_hms = seconds_to_period(duracion_viaje))

head(tabla_general_10)

tipo_bicicleta      fecha_hora_inicio   fecha_hora_fin     
1 bicicleta clásica   2023-02-01 00:03:23 2023-02-01 00:08:42
2 bicicleta docked    2023-02-01 01:22:35 2023-02-01 01:37:44
3 bicicleta docked    2023-02-01 01:39:55 2023-02-01 01:49:01
4 bicicleta clásica   2023-02-01 04:53:51 2023-02-01 04:55:57
5 bicicleta eléctrica 2023-02-01 04:56:34 2023-02-01 05:02:02
6 bicicleta clásica   2023-02-01 05:36:31 2023-02-01 05:52:48
  estacion_inicio                   estacion_fin                     
1 University Ave & 57th St          Ellis Ave & 60th St              
2 Leavitt St & Armitage Ave         Racine Ave & Fullerton Ave (Temp)
3 Racine Ave & Fullerton Ave (Temp) Clark St & Wrightwood Ave        
4 Wabash Ave & Roosevelt Rd         Wabash Ave & 9th St              
5 Wabash Ave & 9th St               Michigan Ave & Washington St     
6 Sheridan Rd & Buena Ave           Southport Ave & Roscoe St        
  latitud_inicio longitud_inicio latitud_fin longitud_fin tipo_cliente
1 41.79148       -87.59986       41.78510    -87.60107    casual      
2 41.91781       -87.68244       41.92551    -87.65840    casual      
3 41.92551       -87.65840       41.92955    -87.64312    casual      
4 41.86723       -87.62596       41.87077    -87.62573    casual      
5 41.87059       -87.62575       41.88398    -87.62468    casual      
6 41.95849       -87.65497       41.94374    -87.66402    casual      
  duracion_viaje duracion_viaje_hms
1 319 secs       5M 19S            
2 909 secs       15M 9S            
3 546 secs       9M 6S             
4 126 secs       2M 6S             
5 328 secs       5M 28S            
6 977 secs       16M 17S

2. Cambie el formato a HH:MM:SS:

In [66]:
tabla_general_10 <- tabla_general_10 %>%
  mutate(duracion_viaje_formateado = sprintf("%02d:%02d:%02d",
                                            hour(duracion_viaje_hms),
                                            minute(duracion_viaje_hms),
                                            second(duracion_viaje_hms)
                                            )
        ) 

head(tabla_general_10)


tipo_bicicleta      fecha_hora_inicio   fecha_hora_fin     
1 bicicleta clásica   2023-02-01 00:03:23 2023-02-01 00:08:42
2 bicicleta docked    2023-02-01 01:22:35 2023-02-01 01:37:44
3 bicicleta docked    2023-02-01 01:39:55 2023-02-01 01:49:01
4 bicicleta clásica   2023-02-01 04:53:51 2023-02-01 04:55:57
5 bicicleta eléctrica 2023-02-01 04:56:34 2023-02-01 05:02:02
6 bicicleta clásica   2023-02-01 05:36:31 2023-02-01 05:52:48
  estacion_inicio                   estacion_fin                     
1 University Ave & 57th St          Ellis Ave & 60th St              
2 Leavitt St & Armitage Ave         Racine Ave & Fullerton Ave (Temp)
3 Racine Ave & Fullerton Ave (Temp) Clark St & Wrightwood Ave        
4 Wabash Ave & Roosevelt Rd         Wabash Ave & 9th St              
5 Wabash Ave & 9th St               Michigan Ave & Washington St     
6 Sheridan Rd & Buena Ave           Southport Ave & Roscoe St        
  latitud_inicio longitud_inicio latitud_fin longitud_fin tipo_cliente
1 41.79148       -87.59986       41.78510    -87.60107    casual      
2 41.91781       -87.68244       41.92551    -87.65840    casual      
3 41.92551       -87.65840       41.92955    -87.64312    casual      
4 41.86723       -87.62596       41.87077    -87.62573    casual      
5 41.87059       -87.62575       41.88398    -87.62468    casual      
6 41.95849       -87.65497       41.94374    -87.66402    casual      
  duracion_viaje duracion_viaje_hms duracion_viaje_formateado
1 319 secs       5M 19S             00:05:19                 
2 909 secs       15M 9S             00:15:09                 
3 546 secs       9M 6S              00:09:06                 
4 126 secs       2M 6S              00:02:06                 
5 328 secs       5M 28S             00:05:28                 
6 977 secs       16M 17S            00:16:17